main.tf

```diff
@@ -223,9 +247,12 @@ data "aws_iam_policy_document" "resource_full_access" {
 }
 
 data "aws_iam_policy_document" "resource" {
-  count                     = module.this.enabled ? 1 : 0
-  source_policy_documents   = local.principals_readonly_access_non_empty ? [data.aws_iam_policy_document.resource_readonly_access[0].json] : [data.aws_iam_policy_document.empty[0].json]
-  override_policy_documents = local.principals_full_access_non_empty ? [data.aws_iam_policy_document.resource_full_access[0].json] : [data.aws_iam_policy_document.empty[0].json]
+  count                   = module.this.enabled ? 1 : 0
+  source_policy_documents = local.principals_readonly_access_non_empty ? [data.aws_iam_policy_document.resource_readonly_access[0].json] : [data.aws_iam_policy_document.empty[0].json]
+  override_policy_documents = [
```

Comment:
```suggestion
  override_policy_documents = distinct([])
```

Reply:
You mean to wrap `distinct([` around lines 253 and 254 ?

Reply:
Yes

Reply:
Also adding lambda principal policy as whole would be great so if someone wants a repo with lambda ran can be done as well without others statements policy